src/uwtools/drivers/driver.py

```diff
@@ -174,43 +197,126 @@ def _taskname(self, suffix: str) -> str:
 
         :param suffix: Log-string suffix.
         """
-        return "%s %s" % (self._driver_name, suffix)
+        cycle = getattr(self, "_cycle", None)
+        leadtime = getattr(self, "_leadtime", None)
+        timestr = (
+            (cycle + leadtime).strftime("%Y%m%d %H:%M:%S")
+            if cycle and leadtime is not None
+            else cycle.strftime("%Y%m%d %HZ") if cycle else None
+        )
+        return " ".join(filter(None, [timestr, self._driver_name, suffix]))
 
-    def _taskname_with_cycle(self, cycle: datetime, suffix: str) -> str:
+    def _validate(self) -> None:
         """
-        Returns a common tag for graph-task log messages.
+        Perform all necessary schema validation.
+        """
+        schema_name = self._driver_name.replace("_", "-")
+        validate_internal(schema_name=schema_name, config=self._config)
 
-        :param suffix: Log-string suffix.
+
+class AssetsCycleBased(Assets):
```

Comment:
Here and for a bit below, new base classes for cycle-based, cycle-and-leadtime-based, and time-invariant drivers, both assets-style and standalone-style. We need these classes for their explicit argument lists: While no interfaces would break if, for example, `ChgresCube` inherited directly from `Assets`, in e.g. the REPL a `help(ChgresCube)` would show the signature and docstring from the `Assets` class' constructor, which includes optional `cycle` and `leadtime` arguments, which aren't valid for `ChgresCube`. So, while a user could still instantiate a `ChgresCube` object by just not providing `cycle` and `leadtime` arguments, it would be confusing to see them there as options. So, these intermediate classes provide correct signatures with only the actually-supported arguments for each driver inheriting from them.